taichi/codegen/llvm/llvm_codegen_utils.cpp

```diff
@@ -6,12 +6,55 @@ namespace lang {
 std::string type_name(llvm::Type *type) {
   std::string type_name_str;
   llvm::raw_string_ostream rso(type_name_str);
-  type->print(rso);
+  type->print(rso, /*IsForDebug=*/false, /*NoDetails=*/true);
   return type_name_str;
 }
 
+/*
+ * Determine whether two types are the same
+ * (required type is required renamed version of the other one) based on the
+ * type name. Check recursively if the types are function types.
+ */
+bool is_same_type(llvm::Type *required, llvm::Type *provided) {
```

Comment:
Would be nice if we can make some comments on a summary of the renaming rule (examples are more than welcomed~)